### DISCUSSION TITLE
Cerficação do software - "Anulação de Factura"

### DISCUSSION THREAD

**Discussion by MateusCazola1998** [assoft-portugal/SAF-T-AO#223] [Q&A] [answered]
Presados estou na fase de envio dos requisitos solicitados pela AGT para certificação do software pelo que um dos pontos deixou-me em dúvidas a anulação de uma factura é feita por meio de uma nota de crédito ? 

![requisito](https://github.com/assoft-portugal/SAF-T-AO/assets/44972898/a6c7e1b8-26c5-43b1-b64c-596453219c4d)

Sobre quais circunstâncias a factura poderá ser anulada sem que obrigatóriamente seja emitida uma nota de crédito.

Desde já obrigado pela atenção.

**Comment by cryptolopes** [comment, marked as answer]
Uma forma de anular um documento de faturação no SAF-T (AO) é alterar o valor associado ao elemento `InvoiceStatus` para `A`, de modo a evidenciar que foi anulado. Esta opção deve ser usada quando o documento foi emitido por engano ou contém erros que impedem a sua comunicação à AGT. Só se pode usar este método caso o destinatário do documento não o tenha em sua posse. Todavia, para além do InvoiceStatus é igualmente necessário refletir as respetivas alterações em `InvoiceStatusDate` e `SourceID`.

Outra forma de anular um documento de faturação no SAF-T (AO) é emitir uma Nota de Crédito que refere o documento original e que anula o seu valor total ou parcial. Esta opção deve ser usada quando o documento vai ser ou já foi comunicado à AGT e é necessário corrigir a respetiva situação tributária.

Em ambos os casos, é necessário indicar o motivo da anulação no campo `References` do documento anulado via Nota de Crédito, bem como no campo `Reason` do elemento `DocumentStatus`.

**Reply by MateusCazola1998** [reply]
Muito Obrigado pelo esclarecimento @cryptolopes

**Reply by MateusCazola1998** [reply]
Os ficheiro SAFT que hoje são comunicados a AGT são os de Facturação e Aquisição de Bens e serviços. 
Para validação do software posso igualmente enviar apenas estes dois tipos de SAFT ou tenho que obrigatóriamente enviar exemplares de todos os tipos de SAFT possíveis de serem emitidos?
